lldb/examples/synthetic/libcxx.py

```diff
@@ -74,6 +77,40 @@ def stdstring_SummaryProvider(valobj, dict):
             return '"' + strval + '"'
 
 
+def get_buffer_data(parent):
+    map_valobj = parent.valobj.GetChildMemberWithName("__map_")
+    map_data = map_valobj.GetChildMemberWithName("__data_")
+    if map_data.IsValid():
+        return map_data
+
+    return map_valobj
+
+
+def get_buffer_end(buffer, begin):
+    map_end = buffer.GetChildMemberWithName("__end_")
+    if map_end.IsValid():
+        return map_end.GetValueAsUnsigned(0)
+    map_size = buffer.GetChildMemberWithName("__size_").GetValueAsUnsigned(0)
```

Comment:
I assume this branch gets hit for the new layouts?

In other words, with this formatter patch, debugging code compiled against an older libc++ wouldn't break right?

Reply:
If so, could we leave comments in the places where we do something for old vs. new layouts?